### PR TITLE
Closes #4536: Add View.findViewHierarchy()

### LINKED ISSUE #4536: Content fixed to the bottom of the screen is behind the URL bar
**Steps to reproduce**

* Run either Reference Browser, or Fenix with [this patch](https://github.com/csadilek/fenix/commit/b48310092ac7a56ab51930356879929fdb024444) applied to enable the dynamic toolbar (URL bar that hides when you scroll).
* Load https://www.sozone.de/.

**Expected results**

The page's green bottom bar (cookie banner) is just above the URL bar.

**Actual results**

The page's green bottom bar (cookie banner) is behind the URL bar.

This is a regression since the time [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1552608) was filed in May.

At the implementation level, it looks like GeckoView's `setVerticalClipping` API, which was added in [bug 1516048](https://bugzilla.mozilla.org/show_bug.cgi?id=1516048) and is supposed to adjust the position of bottom-fixed content, is not being called.

As the issue is common to Fenix and R-B, I assume it's an issue in Android Components.

cc @csadilek @pocmo @snorp

### PR DESCRIPTION
**Pull request by psymoon**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/View.kt [hunk 1]
```diff
@@ -105,6 +106,25 @@ fun View.toScope(): CoroutineScope {
...
+        if (view is ViewGroup) {
+            for (i in 0 until view.childCount) {
+                val childView = findView(view.getChildAt(i))
+                if (childView != null) return childView
+            }
+        }
+        return null
+    }
+
+    return findView(this)
```
Comment (pocmo) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/View.kt]: Is the inner `findView()` function needed? Couldn't we call `findViewInHierarchy` internally directly as `view.getChildAt(i).findViewInHierarchy(predicate)`? 🤔

File: components/feature/session/src/main/java/mozilla/components/feature/session/behavior/EngineViewBottomBehavior.kt [hunk 2]
```diff
@@ -37,8 +38,13 @@ class EngineViewBottomBehavior(
         return super.layoutDependsOn(parent, child, dependency)
     }
 
+    /**
+     * Apply vertical clipping to [EngineView]. This requires [EngineViewBottomBehavior] to be set
+     * in the direct descending child of the [CoordinatorLayout].
```
Comment (csadilek) [inline components/feature/session/src/main/java/mozilla/components/feature/session/behavior/EngineViewBottomBehavior.kt]: nit: Would add to this: to be set in/on the [EngineView] or its parent. Must be a direct descending child of [CoordinatorLayout].

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/View.kt [hunk 3]
```diff
@@ -105,6 +106,22 @@ fun View.toScope(): CoroutineScope {
     return scope
 }
 
+/**
+ * Finds and returns the first [View] in the hierarchy matching the given predicate.
```
Comment (csadilek) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/View.kt]: nit: Finds the first a view in the hierarchy, for which the provided predicate is true.

**Comment by psymoon** [comment]
bors r=csadilek

**Pull request merged**